src/transparent-proxy.c

```diff
@@ -38,17 +38,21 @@
 static int build_url (char **url, const char *host, int port, const char *path)
 {
         int len;
+        char *tmp;
 
         assert (url != NULL);
         assert (host != NULL);
         assert (port > 0 && port < 32768);
         assert (path != NULL);
 
         len = strlen (host) + strlen (path) + 14;
-        *url = (char *) safemalloc (len);
-        if (*url == NULL)
+        
+        tmp = (char *) saferealloc (*url, len);
```

Comment:
what's the purpose of letting the allocation go through a temporary var ?

Reply:
The build_url is called by do_transparent_proxy/process_request, the url param is malloc in process_request. If we use safemalloc, there will be memory leak.

Reply:
that's not really answering my question.
please try again.

also please state which OS you use, which LIBC and which architecture.

Reply:
The temporary var is not necessary
`*url = (char *) saferealloc (*url, len);`

Env:
Openwrt/Linux 3.3.8, uClibc-0.9.33.2, MIPS/x86

Reply:
thanks. can you show the output from valgrind or whatever tool that showed you this leak ?

Reply:
Mmm, we are not using tools, we found it casually when we executed `ps` and found the process memory increased persistent. Then we limit the process in config file
`maxclients 1`
`maxspareservers 1`
`maxrequestsperchild 10000`
Start
`admin@admin:/tmp# ps | grep tinyproxy`
` 2274 admin     1088 S    /usr/sbin/tinyproxy -C /etc/tiny.conf`
` 2291 nobody    1088 S    /usr/sbin/tinyproxy -C /etc/tiny.conf`
After we accessed some web pages, it inceased slowly
`admin@admin:/tmp# ps | grep tinyproxy`
` 2274 admin     1088 S    /usr/sbin/tinyproxy -C /etc/tiny.conf`
` 2291 nobody    1176 S    /usr/sbin/tinyproxy -C /etc/tiny.conf`

At last we enabled the debugging code through NDEBUG, found the leaking code...

Reply:
OK. i believe this 2nd leak is legit; in the worst case if you're wrong nothing will break by changing the malloc here to realloc. could you make a new PR based on master with just that change ?

Reply:
OK, plz close this PR, thanks.